src/Datagrid/ProxyQuery.php

```diff
@@ -57,9 +60,6 @@ public function __clone()
         $this->queryBuilder = clone $this->queryBuilder;
     }
 
-    /**
-     * @return Iterator<object>
```

Comment:
wdyt about this alternative @VincentLanglet ? the typehint for execute is already on the parent and it is already iterable<T> so I don't think we need to add this.

Reply:
That's a solution until https://github.com/doctrine/mongodb-odm/pull/2457 indeed.